R/kernel.r

```diff
@@ -242,6 +243,92 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    make_section_header_plain <- function (txt) paste0("# ", txt, ":\n")
+    make_section_header_html <- function (txt) paste0("<h1>", txt, ":</h1>\n")
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Get token under the cursor_pos.
+    # There may be better ways to do that than this.
+    token <- ""
+    for (i in seq(cursor_pos, nchar(code))) {
+        topic_candidate <- utils:::.guessTokenFromLine(code, i)
+        if (!grepl(token, topic_candidate)) break
+        token <- topic_candidate
+    }
+
+    if (nchar(token) == 0) {
+        data <- list()
+        found <- FALSE
+    } else {
+        data <- list()
+
+        class_data <- tryCatch(
+            eval(parse(text = paste0("class(", token, ")"))),
```

Comment:
Yes, `class(token)` or `IRdisplay::prepare_mimebundle(obj)` works. There is no good reason to use `eval()`.
Also `IRdisplay::prepare_mimebundle(help(token))` basically work, but we can't get help documents in the case `token` is `pkg::topic` or `pkg:::topic` form. We need to split it into `pkg` and `topic` manually and I felt it a little bit hassle than using `eval()`. But I'll try to make it simple without `eval()`.